crates/router/src/connector/expresscheckout.rs

```diff
@@ -0,0 +1,618 @@
+
+#[derive(Debug, Clone)]
+pub struct Expresscheckout;
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Expresscheckout
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let mut headers = vec![
+            (
+                headers::CONTENT_TYPE.to_string(),
+                self.get_content_type().to_string(),
+            ),
+            ("x-merchantid".to_string(), req.merchant_id.clone()),
+            ("version".to_string(), String::from("2018-10-25")),
+        ];
+        let auth = expresscheckout::ExpresscheckoutAuthType::try_from(&req.connector_auth_type)?;
+        let auth_header = (
+            headers::AUTHORIZATION.to_string(),
+            format!("Basic {}", auth.api_key),
+        );
+        headers.push(auth_header);
+        Ok(headers)
+    }
+}
+
+impl ConnectorCommon for Expresscheckout {
+    fn id(&self) -> &'static str {
+        "expresscheckout"
+    }
+
+    fn common_get_content_type(&self) -> &'static str {
+        "application/x-www-form-urlencoded"
+    }
+
+    fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
+        connectors.expresscheckout.base_url.as_ref()
+    }
+
+    // fn get_auth_header(&self, auth_type:&types::ConnectorAuthType)-> CustomResult<Vec<(String,String)>,errors::ConnectorError> {
+    //     let auth: expresscheckout::ExpresscheckoutAuthType = auth_type
+    //         .try_into()
+    //         .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+    //     Ok(vec![(headers::AUTHORIZATION.to_string(), auth.api_key)])
+    // }
+}
+
+impl api::Payment for Expresscheckout {}
+
+impl api::PreVerify for Expresscheckout {}
+impl ConnectorIntegration<api::Verify, types::VerifyRequestData, types::PaymentsResponseData>
+    for Expresscheckout
+{
+}
+
+impl api::PaymentVoid for Expresscheckout {}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Expresscheckout
+{
+    fn get_headers(
+        &self,
+        req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let res = req
+            .response
+            .as_ref()
+            .map_err(|_| errors::ConnectorError::FailedToObtainIntegrationUrl)?;
+        if let types::PaymentsResponseData::TransactionResponse { resource_id, .. } = res {
```

Comment:
```suggestion
        let connector_transaction_id = req.request.connector_transaction_id;
```
The request data should not be taken from `response`. Use the `request` instead